src/parser/toAst.ts

```diff
@@ -95,6 +95,12 @@ export const buildAst = (CSS: string, file = ''): SingleStyleAst => {
       return;
     }
 
+    if (atParents.has(rule.parent)) {
+      atParents.add(rule);
+
+      return;
+    }
```

Comment:
that is an extremely simple solution, but 🤷‍♂️ all tests are happy with it, so do I.

> The better test is to add `.b` to the styles to indicate a (unproven) defect of extracting unnecessary selectors, but the current solution is free from production defect, which is much more important than reducing extraction size.